### ISSUE TITLE
Output [object Object] when trying to identify 'body' parameter in path's apiDoc

### ISSUE THREAD

**Issue by BkunS** [kogosoftwarellc/express-openapi#86]
I was trying to do a PUT /inventories/{id} endpoint that will accept request body as: 
`{ "inventory": 100 }`

I defined the apiDoc in each path js file, and this is the one for PUT.
```
// paths/inventories/{id}.js
const inventoryByIdPath = (inventoriesService) => {
  const PUT = (req, res) => {
    const id = _.get(req, 'params.id');
    const data = _.get(req, 'params.data');
    ...
  };
  PUT.apiDoc = {
   ...
    parameters: [
      {
        in: 'path',
        name: 'id',
        description: 'productId',
        required: true,
        type: 'string'
      },
      {
        in: 'body',
        name: 'data',
        description: 'Inventory Data',
        required: true,
        schema: {
          $ref: '../../configs/schemas.json#/Inventory'
        }
      }
  ]
  ...
}
```

// schemas.json:
```
{
  "Inventory": {
    "properties": {
      "inventory": {
        "type": "number"
      }
    }
  }
}
```

However, when I call this endpoint, it will always return html error response instead of json:
```
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <title>Error</title>
    </head>
    <body>
        <pre>[object Object]</pre>
    </body>
</html>
```

and console would have logged: `[object Object]` after I send the request, which I didn't log it anywhere in code.
After setting the break point, I found out it didn't even make it to the line: `const PUT = (req, res) => {...}`

Though after removing:
```
{
  in: 'body',
  name: 'data',
  description: 'Inventory Data',
  required: true,
  schema: {
    $ref: '../../configs/schemas.json#/Inventory'
  }
}
```
It would work normal, but it won't have 'body' in req.

I tried to find more similar examples online, but only found those provided in express-openapi docs.
Having no clue where would it go wrong. Any ideas? And is there anyway I could know what's inside that mystery [object, Object]?

Using these packages:  
```
{
    "express": "^4.16.2",
    "express-openapi": "^1.2.0"
}
```

**Comment by jsdevel** [comment]
_.get(req, 'body.data')?

**Comment by BkunS** [comment]
There's no 'body' inside req, though '_heepMessage.ServerResponse._hasBody' in req is true.

**Comment by alrik** [comment]
Have you set up the middleware for consuming json?

```
var bodyParser = require('body-parser');
openapi.initialize({
  /*...*/
  consumesMiddleware: {
    'application/json': bodyParser.json(),
    'text/text': bodyParser.text()
  }
  /*...*/
});
```

**Comment by BkunS** [comment]
Tried, but no luck.

**Comment by MugeSo** [comment]
@BkunS See https://github.com/kogosoftwarellc/express-openapi#argsexternalschemas

**Comment by BkunS** [comment]
@MugeSo I'm defining the apiDoc in each path's js file, where would be relative path for args.docPath?

**Comment by MugeSo** [comment]
@BkunS 
First, the content of `$ref` should  be relative path from `args.docPath` because it will be sent as-is to api users as a part of the api doc.
And the definition file referred by `$ref` should be served with resolved URL (args.docPath + $ref).

`arg.externalSchemas` is just used for internally (validation and coercion).

**Comment by BkunS** [comment]
@MugeSo If I have project stucture as follow:
So the args.docPath should be where api-doc.js located, right?

```
project
│   app.js
│   api-doc.js    
└───configs
     └───schemas.json
└───paths
│   └───inventories
│       └─── {id}.js
```
```
// app.js
const apiDoc = require('./api-doc');
openapi.initialize({
  app,
  apiDoc: apiDoc,
  paths: [
    path.resolve(__dirname, 'paths'),
  ],
  ...
});
```
```
// api-doc.js:
const apiDoc = {
  ...
  definitions: {
    Inventory: {
        type: 'number'
    }
  }
}
module.exports = apiDoc
```
```
// paths/inventories/{id}.js:
const inventoryByIdPath = (inventoriesService) => {
  const PUT = (req, res) => {...}
  PUT.apiDoc = {
    ...
    parameters: [
      {
        in: 'path',
        name: 'id',
        description: 'productId',
        required: true,
        type: 'string'
      },
      {
        in: 'body',
        name: 'data',
        description: 'Inventory Data',
        required: false,
        schema: {
          $ref: './configs/schemas.json#/Inventory'
        }
      }
    ]
  }
  ...
}
```
So far, besides `$ref: './configs/schemas.json#/Inventory'`, I've also tried `$ref: '#/definitions/Inventory`, `$ref: '#/apiDoc/definitions/Inventory'`, but none of them works. It still gives me `[object, Object]` error.

**Comment by alrik** [comment]
I think i traced down the issue.

**TLDR;**: you might have **no error handling middleware** defined

---

I was able to reproduce this with the following 4 lines of code:

```js
const express = require('express');
const app = express();

app.get('/', (req, res, next) => next({ status: 400 }) );
app.listen(9095, () => console.warn('Started'));
```
Launch http://localhost:9095 and see `[object Object]`.

The same happens with express-openapi when a validation fails. The validator internally triggers `next({ status : 400, ...someValidationErrorInfo })` in case of a failed validation. When you have no [express error handler](http://expressjs.com/en/guide/error-handling.html) `(err, req, res, next) => {}`  
or have not defined an [express-openapi errorMiddleware](https://github.com/kogosoftwarellc/express-openapi#argserrormiddleware) express tries to convert the error in your case something like `{ status: 400, message: 'validation failed' }` to a string and send it as error response.

**Comment by BkunS** [comment]
@alrik @jsdevel Thanks! With the error handling middleware, I was able to know what's inside the error now, and realized that this whole thing is just because I also didn't use body-parser which I once thought was built in with express. This can be closed.
